Document zstd compression support for log archives (#36379)
DataDog/documentation: content/en/logs/log_configuration/archive_search.md

* **Set Scan Limits:** Admins with `Logs Write Archives` permissions can set a maximum scan size per Archive in the settings.
* **Use Partition Attributes (Preview):** The most effective way to accelerate searches on low-cardinality data like `service`, `env`, or `status`. Datadog skips entire partitions that don't match your query.
* **Use Lookup Attributes (Preview):** The most effective way to accelerate searches on high-cardinality data like `trace_id` or `user_id`.

> * **Use zstd compression:** Archives use zstd compression by default, which offers better compression ratios than gzip and reduces the volume of data scanned, lowering cloud egress costs. If your archive is configured to use gzip, consider switching to zstd. See [Log Archives][9] to configure compression.

Reviewer: ```suggestion
* **Use zstd compression:** Archives use zstd compression by default, which reduces scan volume and cloud egress costs compared to gzip. If your archive uses gzip, see [Log Archives][9] to switch to zstd.
```
